### PR TITLE
feat: auto generate file

### PR DESCRIPTION
**Pull request by anhle10051996**

### CODE REVIEW

File: template/package.json [hunk 1]
```diff
@@ -17,13 +17,15 @@
...
-    "generatefonts": "node scripts/fonts"
+    "generatefonts": "node scripts/fonts",
+    "plop": "plop"
   },
   "dependencies": {
     "@react-navigation/native": "^6.0.13",
     "@react-navigation/native-stack": "^6.9.1",
     "@reduxjs/toolkit": "^1.8.6",
     "i18n-js": "^4.1.1",
+    "plop": "^3.1.1",
```
Comment (honghoangsts) [inline template/package.json:28]: This library should add to `devDependencies` instead.

**Review by honghoangsts — CHANGES_REQUESTED** [review]

File: template/plopfile.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+        choices: ['screens', 'component', 'store'],
+      },
+    ],
+    actions(data) {
+      switch (data.type) {
+        case 'screens':
+          return screensView
+        case 'store':
+          return store
+        case 'component':
```
Comment (honghoangsts) [inline template/plopfile.js:139]: Should we create constant for this?

**Comment by honghoangsts** [comment]
There are a lot of comment? Do we need it? Can we remove it?

**Review by honghoangsts — CHANGES_REQUESTED** [review]

**Review by baonguyenhsts — APPROVED** [review]

**Review by huydosgtech — APPROVED** [review]

File: template/plopfile.js (hunk 2 above)
Comment (anhle10051996) [reply template/plopfile.js]: Fixed

File: template/package.json (hunk 1 above)
Comment (anhle10051996) [reply template/package.json]: Updated

**Review by vunguyendhSTS — APPROVED** [review]

File: template/plopfile.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
+const module_type = {
```
Comment (huydosgtech) [inline template/plopfile.js:1]: wrong format

**Review by huydosgtech — APPROVED** [review]

**Review by honghoangsts — APPROVED** [review]

**Pull request merged**